include/thread.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2016, Wind River Systems, Inc.
+ * Copyright (c) 2020, Intel Coporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_THREAD_H_
```

Comment:
suggestion: put all this in include/sys/thread.h, to at least space this under sys/ in the include space rather than top-level. I think we want to avoid any new toplevel headers.